packages/gestalt/src/SideNavigation/NavigationContent.js

```diff
@@ -52,6 +54,7 @@ export default function NavigationContent({
     const mouseEnterHandler = () => {
       if (sideNavigationCollapsed && !transitioning) {
         clearTimeout(previewTimeoutRef.current);
+        setCollapsedContainerWidth(mainContainer.current?.offsetWidth);
```

Comment:
By default the main wrapper has no width set. 
Its "calculated/rendered" width is `contentWidth + scrollbars`. 
We need to record this just before we expand the sidenav as overlay. So that it doesn't push the page content. The content of the sidenav will still expand overflowing the main wrapper.